src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs

```diff
@@ -91,33 +91,28 @@ internal OtlpLogs.LogRecord ToOtlpLog(LogRecord logRecord)
             var attributeValueLengthLimit = this.sdkLimitOptions.LogRecordAttributeValueLengthLimit;
             var attributeCountLimit = this.sdkLimitOptions.LogRecordAttributeCountLimit ?? int.MaxValue;
 
-            /*
-            // Removing this temporarily for stable release
-            // https://github.com/open-telemetry/opentelemetry-dotnet/issues/4776
-            // https://github.com/open-telemetry/opentelemetry-dotnet/issues/3491
-            // First add the generic attributes like Category, EventId and Exception,
-            // so they are less likely being dropped because of AttributeCountLimit.
-
-            if (!string.IsNullOrEmpty(logRecord.CategoryName))
+            if (this.experimentalOptions.EmitLogEventAndCategoryAttributes)
             {
-                // TODO:
-                // 1. Track the following issue, and map CategoryName to Name
-                // if it makes it to log data model.
-                // https://github.com/open-telemetry/opentelemetry-specification/issues/2398
-                // 2. Confirm if this name for attribute is good.
-                otlpLogRecord.AddStringAttribute("dotnet.ilogger.category", logRecord.CategoryName, attributeValueLengthLimit, attributeCountLimit);
-            }
+                if (!string.IsNullOrEmpty(logRecord.CategoryName))
+                {
+                    // TODO:
+                    // 1. Track the following issue, and map CategoryName to Name
+                    // if it makes it to log data model.
+                    // https://github.com/open-telemetry/opentelemetry-specification/issues/2398
+                    // 2. Confirm if this name for attribute is good.
+                    AddStringAttribute(otlpLogRecord, "dotnet.ilogger.category", logRecord.CategoryName, attributeValueLengthLimit, attributeCountLimit);
+                }
 
-            if (logRecord.EventId.Id != default)
-            {
-                otlpLogRecord.AddIntAttribute(nameof(logRecord.EventId.Id), logRecord.EventId.Id, attributeCountLimit);
-            }
+                if (logRecord.EventId.Id != default)
+                {
+                    AddIntAttribute(otlpLogRecord, "event.id", logRecord.EventId.Id, attributeCountLimit);
```

Comment:
"dotnet.ilogger.event.id" have we considered this, given we prefix "dotnet.ilogger" to the category name?

Reply:
Category name is still open for discussion. https://github.com/open-telemetry/opentelemetry-dotnet/issues/3491.

I do not have strong take on this yet. If we decide to use `InstrumentationScope` for category then do we need still need this?

Something to be discussed on the issue. Comments are misleading in code (I will fix it).